src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -149,7 +150,15 @@ public void lockShared(UUID datasetId, String flightId) {
         MapSqlParameterSource params = new MapSqlParameterSource()
             .addValue("datasetid", datasetId)
             .addValue("flightid", flightId);
-        int numRowsUpdated = jdbcTemplate.update(sql, params);
+        int numRowsUpdated = 0;
+        try {
+            numRowsUpdated = jdbcTemplate.update(sql, params);
+        } catch (DataAccessException dataAccessException) {
+            if (retryQuery(dataAccessException)) {
+                throw new RetryQueryException("Retry");
+            }
+            throw new FatalQueryException("Don't Retry");
```

Comment:
Can you nest (i.e. call the other constructor that takes a throwable also) the original dataAccessException in these new exceptions, so that we don't lose the root cause?

Also, I like the RetryQueryException because it allows us to group related PostGres/JDBC exception together that have similar behavior (i.e. retryable) in stairway. But do we need the FatalQueryException? I think it's just adding an extra wrapper over what we already have -- all exceptions are fatal by default.

Reply:
Thank you for the helpful comments! I've changed RetryQueryException to inherit from DataAccessException and removed the FatalQueryException.

Reply:
Ah, ok, thank you! Changes made.